session/create_consumer.go

```diff
@@ -70,14 +80,33 @@ func (consumer *createConsumer) Consume(requestPtr interface{}) (response interf
 				destroyCallback()
 			}()
 		}
-		return responseWithSession(sessionInstance, config, nil), nil
+		return responseWithSession(sessionInstance, config, consumer.loadPaymentInfo(issuerID)), nil
```

Comment:
Communication consumer/provider is meant to be thin in term of business logic, it's just a transport layer with mapping to communication DTOs. So then we can refactor them easily later.

Back in a day I followed MVC principle *thin Controller, fat Service* here:

- `createConsumer` - is meant to be a Controller here. So thin and just calls `consumer.SessionManager.Create()`
- `SessionManager. sessionInstance` - is meant to be a Service

Reply:
We have 2 services, the lookup of last promise and session creation. They are called, and the response is remapped to a dto.

Reply:
Still not fixed, we need to make `SessionManager.Create()` the only class to make connections in network